aiosmtpd/handlers.py

```diff
@@ -89,15 +89,15 @@ def __init__(self, remote_hostname, remote_port):
         self._port = remote_port
 
     def process_message(self, peer, mailfrom, rcpttos, data, **kws):
-        lines = data.split('\n')
+        lines = data.splitlines(keepends=True)
         # Look for the last header
         i = 0
         for line in lines:                          # pragma: nobranch
-            if not line:
+            if line in ('\r', '\n', '\r\n'):
                 break
             i += 1
-        lines.insert(i, 'X-Peer: %s' % peer[0])
-        data = NEWLINE.join(lines)
+        lines.insert(i, 'X-Peer: %s\r\n' % peer[0])
```

Comment:
I don't like unconditionally adding the `\r` to `X-Peer`.  If `\n` was the only line ending seen in the data, then I'd like for `\n` to be used for the `X-Peer` line.

What do you think about keeping track of the line breaks that you see as you're iterating over them?  Then if you see all of one kind, you'd use that to terminate the `X-Peer` line?  If you see a mix, then yes, use `\r\n`.